containerd-shim-spin-config.md

```diff
@@ -0,0 +1,372 @@
+own configuration from these containerd runtime options.
+
+The following are Spin shim and Runwasi options that should be configurable:
+
+| Key                             | Example Value | Scope                                                                                                                                                           |
+|---------------------------------|---------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| DISABLE_PRECOMPILATION          | "true"        | shim                                                                                                                                                            |
+| GRPC_MAX_WRITE_CHUNK_SIZE_BYTES | "15728640"    | [Runwasi](https://github.com/containerd/runwasi/blob/7cc5e5a8025f1508c264dee1faf1c8c57cf645a1/crates/containerd-shim-wasm/src/sandbox/containerd/client.rs#L38) |
+
+Ideally, these options could be set in the `SpinAppExecutor` CRD. Then, the
+RuntimeClassManager can watch the resource and update and restart containerd as
+needed.
+
+### Spin Shim Executor Environment Variables
+
+Since containerd shims inherit the environment variables of the calling
+containerd process, environment variables that need to be set on the shim
+process need to be set on containerd. Configuring containerd environment
+variables differs depending on how the service is executed. When directly
+invoking containerd, the log level of the shim could be adjusted as follows:
+
+```sh
+RUST_LOG=info containerd --address /tmp/containerd.sock
+```
+
+If the process is managed by systemd, you can add an environment variable to the
+systemd service file. First, stop the service and bring up the service file:
+
+```sh
+systemctl stop containerd
+systemctl edit containerd
+```
+> note, for k3s, edit the k3s service (`systemctl edit k3s`)
+
+At the top of the file, add the environment variable to the service:
+
+```sh
+[Service]
+Environment="RUST_LOG=info"
+```
+
+Then the containerd service must be restarted to surface the change.
+
+```sh
+systemctl restart containerd
+```
+
+For this specific scenario, a log level field could be added to the
+`SpinAppExecutor` and the Runtime Class Manager could handle configuring and
+restarting containerd.
+
+## References
+
+- Runwasi issue for tracking containerd config options for shims:
+  https://github.com/containerd/runwasi/issues/573
+- `containerd-shim-spin` issue tracking inability to run two Spin apps in the
+  same container due to fixed listen address:
+  https://github.com/spinkube/containerd-shim-spin/issues/52
+- `containerd-shim-spin` issue considering which `spin up` flags to support in
+  the shim: https://github.com/spinkube/containerd-shim-spin/issues/130
+
+## Alternatives Considered
+
+### Configuring Spin Execution Alternatives
```

Comment:
Can you add a section here that describes passing all the ENVs to all components?  We've discuss and changed that a few times so should document why we discarded it as a solution